### ISSUE TITLE
Which are pros and cons of monolithics applications?

### ISSUE THREAD

**Issue by wbigger** [wbigger/2018-19-5Binf#7]
Please answer to this issue with advantages (pros) and disadvantages (cons) of monolithcs applications, according to you.

English and Italian are both fine.

**Comment by pastax** [comment]
you can't force the user to perform an update

**Comment by micia01** [comment]
A  ** pro** of a monolithic application is that it can run even if there's no internet connection, but a **con**  is that if there's some upgrades you have to download and install them but if you don't want to do it you will have an old application, maybe with some bugs 😌

**Comment by wbigger** [comment]
Yes! What about your application is CPU/GPU intensive, ie. 3D rendering? Do you think it is a good think to have everything local on your machine?

**Comment by Pierluigi22** [comment]
Yes, I think it's a good think, beacuse you can keep an eye on your whole house, for example, on any device without go around the house to check everything.
Anyway, I think a pro is that updates can improve application security, while, a con is that the application could have minium requirements (for example, the 3d render, which some devices may not support due to incompatible hardware).

**Comment by manuellomartire** [comment]
@pastax I think instead that you have to force the user to do an update of an application because there maybe bugs that can be very dangerous like for example burn the pc for a stupid bug or lose all data😂

**Comment by wbigger** [comment]
Good point @manuellomartire . The kind of application should be taken into consideration, eg. how critical is the application and the possible risks.

**Comment by 0Shaeid0** [comment]
>           Please answer to this issue with advantages (pros) and disadvantages (cons) of monolithcs applications, according to you.
> English and Italian are both fine.

The pros and the cons depends on what you need to do.
If you want a big application, which needs a lot of requirements, a monolithc application isn't the best solution, but if you need something small and not very powerful, then can be a solution.

**Comment by ronny1926** [comment]
**The monolithic app** like pro has the use without internet, which makes it easily transportable but above all using it using few resources.
**1.** But this can lead to the presence of spyware that will never be deleted as the app has no obligation to update . The first is against this.
**2.**The second is the lack of resolution of the bugs that could lead to problems related to the security of privacy and personal data. @wbigger

**Comment by wbigger** [comment]
> The pros and the cons depends on what you need to do.
> If you want a big application, which needs a lot of requirements, a monolithc application isn't the best solution, but if you need something small and not very powerful, then can be a solution.

Application size is a factor but probably not the main one. Very small applications may be find their natural way as a distributed application, and the other way around big application may find theri best implementation a single tier application.
